Refactor flashprog

funkeleinhorn opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

[Flashprog](https://flashprog.org) is a fork of flashrom. It would be nice for users to be able to choose which of them they would like to use. Also flashprog has some features flashrom does not have (yet) like support for multiple SPI chip selects.

based on pkgs/tools/misc/flashrom/default.nix
can also be used with the nixos option
```nix
programs.flashrom = {
  enable = true;
  package = pkgs.flashprog;
};
```

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [X] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [X] `sandbox = relaxed`
  - [X] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

funkeleinhorn committed:
maintainers: add funkeleinhorn

funkeleinhorn committed:
flashprog: refactor make jlink and gpio support configurable

hcsch reviewed (approved):
LGTM. As this is based on the flashrom package it's already likely to work, since flashprog is a fork of that. I've also tested the build of the package locally and it produces a working executable on current nixos-unstable on x86_64-linux :)

hcsch commented:
With the force-push things seem to have broken with some now missing dependencies. Somehow managed to build the code before the force-push. Will review again once that's fixed

hcsch reviewed (approved):
Alright, now it builds again and actually works with what is currently pushed here (ef9362ce125bf6facef9bc32c3680293868bb62a)

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/1483

AndersonTorres commented on pkgs/by-name/fl/flashprog/package.nix:
```diff
@@ -0,0 +1,53 @@
+  nativeBuildInputs = [ pkg-config installShellFiles ];
+  buildInputs = [ libftdi1 libusb1 ]
+    ++ lib.optionals (!stdenv.isDarwin) [ pciutils ]
+    ++ lib.optional jlinkSupport libjaylink
+    ++ lib.optional gpioSupport libgpiod;
```
```suggestion
    ++ lib.optionals jlinkSupport [ libjaylink ]
    ++ lib.optionals gpioSupport [ libgpiod ];
```

Let's be consistent.

AndersonTorres commented on pkgs/by-name/fl/flashprog/package.nix:
```diff
@@ -0,0 +1,53 @@
+      --replace 'GROUP="plugdev"' 'TAG+="uaccess", TAG+="udev-acl"'
+  '';
+
+  makeFlags = [ "PREFIX=$(out)" "libinstall" ]
+    ++ lib.optionals (!jlinkSupport) [ "CONFIG_JLINK_SPI=no" ]
+    ++ lib.optionals (!gpioSupport) [ "CONFIG_LINUX_GPIO_SPI=no" ]
+    ++ lib.optionals stdenv.isDarwin [ "CONFIG_ENABLE_LIBPCI_PROGRAMMERS=no" ]
+    ++ lib.optionals (stdenv.isDarwin && stdenv.isx86_64) [ "CONFIG_INTERNAL_X86=no" "CONFIG_INTERNAL_DMI=no" "CONFIG_RAYER_SPI=no" ];
```
```suggestion
  makeTargets = [ "libinstall" ];
  
  makeFlags = [
    "PREFIX=$(out)"
  ] ++ lib.optionals (!jlinkSupport) [
    "CONFIG_JLINK_SPI=no" 
  ] ++ lib.optionals (!gpioSupport) [
    "CONFIG_LINUX_GPIO_SPI=no" 
  ] ++ lib.optionals stdenv.isDarwin [ 
    "CONFIG_ENABLE_LIBPCI_PROGRAMMERS=no" 
  ] ++ lib.optionals (stdenv.isDarwin && stdenv.isx86_64) [ 
    "CONFIG_INTERNAL_X86=no" 
    "CONFIG_INTERNAL_DMI=no" 
    "CONFIG_RAYER_SPI=no" 
  ];
```

AndersonTorres commented on pkgs/by-name/fl/flashprog/package.nix:
```diff
@@ -0,0 +1,53 @@
+    install -Dm644 util/z60_flashprog.rules $out/lib/udev/rules.d/z60_flashprog.rules
+  '';
+
+  meta = with lib; {
```
```suggestion
  meta = {
```

Nested with is bad.

AndersonTorres commented on pkgs/by-name/fl/flashprog/package.nix:
```diff
@@ -0,0 +1,53 @@
+  meta = with lib; {
+    homepage = "https://flashprog.org";
+    description = "Utility for reading, writing, erasing and verifying flash ROM chips";
+    license = licenses.gpl2;
+    maintainers = with maintainers; [ funkeleinhorn ];
+    platforms = platforms.all;
+    mainProgram = "flashprog";
```
```suggestion
    license = with lib.licenses; [ gpl2Plus ];
    mainProgram = "flashprog";
    maintainers = with lib.maintainers; [ funkeleinhorn ];
    platforms = lib.platforms.all;
```

AndersonTorres commented on pkgs/by-name/fl/flashprog/package.nix:
```diff
@@ -0,0 +1,53 @@
+, libusb1
+, pciutils
+, pkg-config
+, jlinkSupport ? false
+, gpioSupport ? false
```
Why not set them as true by default?

Also, let's follow https://github.com/NixOS/rfcs/pull/169 a bit:

```suggestion
, with_jlink ? true
, with_gpio ? true
```

AndersonTorres commented:
> Could this package be backported once its added?

Yes. Let's set the backport Github flag.

funkeleinhorn commented:
I added the suggested changes. Please check if everything is okay now.

funkeleinhorn commented:
Build was failing now as libgpiod is not available on darwin. Added check stdenv.isLinux to fix that.

AndersonTorres reviewed (approved):
LGTM

hcsch reviewed (approved):
Works on x86_64 linux with nixos-unstable, and `nixpkgs-fmt` has nothing to complain about. Looks good to me as well :)

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/1493

funkeleinhorn commented:
As someone else opened a pull request to add flashprog (https://github.com/NixOS/nixpkgs/pull/292592) while this one was already open which got merged earlier I rebased my changes on the current master and added the improvments of this pull request like configurable support for jlink and gpio to the now existing package.

felixsinger commented on pkgs/by-name/fl/flashprog/package.nix:
```diff
@@ -1,49 +1,63 @@
-    platforms = platforms.all;
+    license = with lib.licenses; [ gpl2Plus ];
     mainProgram = "flashprog";
+    maintainers = with lib.maintainers; [ felixsinger funkeleinhorn ];
```
I appreciate your interest in becoming a maintainer for this package. Though, maintainers should work well together, so that conflicts are avoided. I'm open and looking forward to work with you in the future, but for now I rather would like to wait with the maintainership here. If you would like to, let's have a chat.

funkeleinhorn replied:
Okayy. Also looking forward to work with you in the future. I removed myself from the maintainers list for now. I would be open to have a chat.

felixsinger commented on pkgs/by-name/fl/flashprog/package.nix:
```diff
@@ -1,49 +1,63 @@
-    installShellFiles
-    pkg-config
-  ];
+  nativeBuildInputs = [ pkg-config installShellFiles ];
+  buildInputs = [ libftdi1 libusb1 ]
```
Use one line per package

felixsinger reviewed (changes requested):
Thanks for your contribution. Looks good overall. I only had a quick look and I will test your changes a bit later. Please split up your commits by topic and avoid bulk changes in one commit.

funkeleinhorn commented on pkgs/by-name/fl/flashprog/package.nix:
```diff
@@ -1,49 +1,63 @@
+    license = with lib.licenses; [ gpl2Plus ];
     mainProgram = "flashprog";
+    maintainers = with lib.maintainers; [ felixsinger funkeleinhorn ];
+    platforms = lib.platforms.all;
```
The move was included in an suggested change by @AndersonTorres. I changed it back to where it was before.

funkeleinhorn commented:
I applied your suggested changes. Hopefully the changelog looks a bit cleaner now.  Please have a look if everything is fine now :)

felixsinger commented on pkgs/by-name/fl/flashprog/package.nix:
```diff
@@ -24,26 +26,38 @@ stdenv.mkDerivation rec {
-    license = with licenses; [ gpl2 gpl2Plus ];
-    maintainers = with maintainers; [ felixsinger ];
-    platforms = platforms.all;
+    license = with lib.licenses; [ gpl2 gpl2Plus ];
+    maintainers = with lib.maintainers; [ felixsinger ];
+    platforms = lib.platforms.all;
```
It feels like people are going back and forth with the with statement. If the change is just because of personal preference, then please change it back.

funkeleinhorn replied:
@AndersonTorres said in their review that nesting with statements is bad and I should change it to that. But I can also change it back if preferred.

felixsinger reviewed (changes requested):
Thank you! I have yet another comment but otherwise I'm fine.

AndersonTorres commented on pkgs/by-name/fl/flashprog/package.nix:
```diff
@@ -24,19 +26,31 @@ stdenv.mkDerivation rec {
-  makeFlags = [ "PREFIX=$(out)" "libinstall" ]
-    ++ lib.optionals stdenv.isDarwin [ "CONFIG_ENABLE_LIBPCI_PROGRAMMERS=no" ]
-    ++ lib.optionals (stdenv.isDarwin && stdenv.isx86_64) [ "CONFIG_INTERNAL_X86=no" "CONFIG_INTERNAL_DMI=no" "CONFIG_RAYER_SPI=0" ];
+  makeFlags = [
+    "PREFIX=$(out)"
+    "libinstall"
+  ] ++ lib.optionals (!with_jlink) [
+    "CONFIG_JLINK_SPI=no"
+  ] ++ lib.optionals (!(stdenv.isLinux && with_gpio)) [
+    "CONFIG_LINUX_GPIO_SPI=no"
+  ] ++ lib.optionals stdenv.isDarwin [
+    "CONFIG_ENABLE_LIBPCI_PROGRAMMERS=no"
+  ] ++ lib.optionals (stdenv.isDarwin && stdenv.isx86_64) [
+    "CONFIG_INTERNAL_X86=no"
+    "CONFIG_INTERNAL_DMI=no"
+    "CONFIG_RAYER_SPI=no"
+  ];
```
A curiousity:
These Make variables accept "yes" as argument?
If yes then you can rewrite this as

```suggestion
  makeFlags = let
    yesNo = flag: if flag then "yes" else "no";
  in [
    "libinstall"   
    "PREFIX=$(out)"
    "CONFIG_JLINK_SPI=${yesNo with_jlink}"
    "CONFIG_LINUX_GPIO_SPI=${yesNo (stdenv.isLinux && with_gpio)}"
    "CONFIG_ENABLE_LIBPCI_PROGRAMMERS=${yesNo (!stdenv.isDarwin)}"
    "CONFIG_INTERNAL_X86=${yesNo (!(stdenv.isDarwin && stdenv.isx86_64))}"
    "CONFIG_INTERNAL_DMI=${yesNo (!(stdenv.isDarwin && stdenv.isx86_64))}"
    "CONFIG_RAYER_SPI=${yesNo (!(stdenv.isDarwin && stdenv.isx86_64))}"
  ];
```

funkeleinhorn replied:
Yes these variables also accept yes as an argument. @felixsinger should I rewrite the make flags as suggested or do you want to keep them like they are?

AndersonTorres replied:
It has the advantage of being explicit and not relying on hidden defaults - think in the hypothesis that in a future release they mark all such add-ons as "disabled by default".

funkeleinhorn replied:
I also think this changes make sense as it makes the package more robust as it not assumes make defaults. I added the suggested changes to my commit. @felixsinger please have a look if you are okay with them. If not I will revert them.

felixsinger replied:
I liked the state before, because it was easily visible which libraries (don't) work under certain conditions.

On the other hand, not every programmer has or needs its own library, as it is the case for libgpio. Some are based on libpci and others libusb. So the new way gives more control over which specific functions (programmers here) should be enabled and it ensures the actual state of the options. I like that.

I know some cases which use NixOS for their embedded solutions and also use flashprog. That definitely will be beneficial.

AndersonTorres reviewed (commented):
Besides, LGTM.

SuperSandro2000 commented on pkgs/by-name/fl/flashprog/package.nix:
```diff
@@ -24,19 +26,29 @@ stdenv.mkDerivation rec {
     pciutils
+  ] ++ lib.optionals with_jlink [
+    libjaylink
+  ] ++ lib.optionals (stdenv.isLinux && with_gpio) [
```
```suggestion
  ] ++ lib.optionals (with_gpio) [
```
having the option true and still not adding the binary is not great

felixsinger replied:
libgpio is specific to Linux. The isLinux condition needs to stay.

AndersonTorres replied:
This is not the typical code on Nixpkgs.
We usually employ

```nix
{
... 
, enableXYZ? systemHasABC
... 
}:

{
... 
  confstring = if enableXYZ then "--enable-xyz" else "disable-xyz";
... 
} 
```

in such cases.
The other format

```nix
{
... 
, enableXYZ? true
... 
}:

{
... 
  confstring = if (enableXYZ && systemHasABC) then "--enable-xyz" else "disable-xyz";
... 
} 
```

makes `enableXYZ` a no-op when `systemHasABC` is false, violating the intention of the user that sets `enableXYZ`.

If it is so essential and imperative to build with `enableXYZ` only when `systemHasABC` is true, then use an assertion:

```nix
{
... 
, enableXYZ? systemHasABC
... 
}:

assert enableXYZ - > systemHasABC;
{
... 
  confstring = if enableXYZ then "--enable-xyz" else "disable-xyz";
... 
} 
```

funkeleinhorn replied:
I removed the isLinux condition here and set withGpio defaults to true for isLinux and to false otherwise. This way darwin users get an explictit error when setting withGpio to true instead of just ignoring the option. I think this is more transparent to the user.

SuperSandro2000 commented on pkgs/by-name/fl/flashprog/package.nix:
```diff
@@ -2,12 +2,14 @@
 , pciutils
 , pkg-config
 , stdenv
+, with_jlink ? true
+, with_gpio ? true
```
Why do we need those options? You didn't provide a justification for them in the thread.

```suggestion
, withJlink ? true
, withGpio ? stdenv.isLinux
```
also we use camleCase

AndersonTorres replied:
camelCase will be deprecated by RFC 169.

infinisil replied:
I don't think it makes sense to try to adhere to an RFC that isn't accepted

AndersonTorres replied:
OK, makes sense.

AndersonTorres replied:
> Why do we need those options?

It is very typical on Nixpkgs.

felixsinger replied:
Yeah, I don't see an issue here. They allow people configuring the software. As I just wrote in another comment here, I know cases where NixOS and flashprog is used in embedded solutions. Many times flash sizes are small and you are happy about every single byte you can save. These options make it easier for people to customize flashprog to their use cases. I want to keep that.

felixsinger replied:
I think this can be marked as resolved. I can't do that though.

felixsinger replied:
Though, as long as the RFC is not approved yet, we should go with camel case. Please change that.

funkeleinhorn replied:
I changed it to camelCase.

SuperSandro2000 commented on pkgs/by-name/fl/flashprog/package.nix:
```diff
@@ -24,19 +26,29 @@ stdenv.mkDerivation rec {
+    "libinstall"
+    "PREFIX=$(out)"
+    "CONFIG_JLINK_SPI=${yesNo with_jlink}"
+    "CONFIG_LINUX_GPIO_SPI=${yesNo (stdenv.isLinux && with_gpio)}"
```
```suggestion
    "CONFIG_LINUX_GPIO_SPI=${yesNo with_gpio}"
```

felixsinger replied:
Same thing here. libgpio is specific to Linux and the condition needs to stay.

SuperSandro2000 commented on pkgs/by-name/fl/flashprog/package.nix:
```diff
@@ -24,19 +26,29 @@ stdenv.mkDerivation rec {
+    "CONFIG_JLINK_SPI=${yesNo with_jlink}"
+    "CONFIG_LINUX_GPIO_SPI=${yesNo (stdenv.isLinux && with_gpio)}"
+    "CONFIG_ENABLE_LIBPCI_PROGRAMMERS=${yesNo (!stdenv.isDarwin)}"
+    "CONFIG_INTERNAL_X86=${yesNo (!(stdenv.isDarwin && stdenv.isx86_64))}"
+    "CONFIG_INTERNAL_DMI=${yesNo (!(stdenv.isDarwin && stdenv.isx86_64))}"
+    "CONFIG_RAYER_SPI=${yesNo (!(stdenv.isDarwin && stdenv.isx86_64))}"
```
Aren't the outer () optional?

felixsinger replied:
I agree, please remove.

funkeleinhorn replied:
No those are not optional. If you remove them you get an syntax error in the listToAttrs builtin: error: syntax error, unexpected '!', expecting '}' This is why the outer () need to stay as the package won't compile otherwise.

AndersonTorres replied:
> Aren't the outer () optional?

I learned this the bad way: ! almost requires ()

felixsinger reviewed (changes requested):
Sorry for responding a bit late. I've commented on the open threads. I think it comes down to 2 minor issues / nits. So just removing two brackets and moving snake case to camel case.

Otherwise it looks good!

Edit: When changes are suggested, I would appreciate if they are discussed first before they get applied. I feel like it otherwise has lots of potential for creating mess. So please only apply changes when people, and especially the maintainer(s), agree on these.

funkeleinhorn commented on pkgs/by-name/fl/flashprog/package.nix:
```diff
@@ -2,15 +2,17 @@
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation {
```
Removed the `rec` as it is not used right now and my LSP/IDE showed a waring because of it.

funkeleinhorn commented:
Could someone restart the failed job as it seems to be an problem with ofborg instead of my package.

AndersonTorres commented:
@ofborg build flashprog

AndersonTorres commented:
> Could someone restart the failed job as it seems to be an problem with ofborg instead of my package.

You can trick ofborg by rebasing with master and force-pushing.

AndersonTorres commented:
P.S.: fixup meta.license

funkeleinhorn commented:
> P.S.: fixup meta.license

what should be fixed?  the current license was requested by @felixsinger and I think it is correct:
> flashprog is mainly GPLv2 licensed. So change it back to what it was before.

AndersonTorres commented:
"GPL2" is  ambiguous, it can refer to GPL2-Only or GPL2-Plus.
Then it makes few to no sense to make a list containing the ambiguous GPL2 and the inambiguous GPL2Plus.

Either the `gpl2` should be removed or it should be changed to `gpl2Only`.

But I will not pursue this minor issue now.

AndersonTorres commented on pkgs/by-name/fl/flashprog/package.nix:
```diff
@@ -1,49 +1,63 @@
-    license = with licenses; [ gpl2 gpl2Plus ];
-    maintainers = with maintainers; [ felixsinger ];
-    platforms = platforms.all;
+    license = with lib.licenses; [ gpl2Plus ];
```
Gpl2 is ambiguous and not recommended

https://www.gnu.org/licenses/identify-licenses-clearly.html

felixsinger commented:
Clarified license in #298706. Let's try to fix the Ofborg issues and then let's get this in.

funkeleinhorn commented:
I rebased on current master and changed the license to gpl2Plus as flashprog is licensed with gpl2Plus. From flashprogs COPYING:

> This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation; either version 2 of the License, or
    (at your option) **_any later version._**

AndersonTorres commented:
OfBorg complains on ARM64-Linux:

> The following features are unavailable on your machine: CONFIG_INTERNAL_X86=yes CONFIG_INTERNAL_X86=yes CONFIG_RAYER_SPI=yes

felixsinger commented on pkgs/by-name/fl/flashprog/package.nix:
```diff
@@ -24,24 +26,36 @@ stdenv.mkDerivation rec {
+      "CONFIG_ENABLE_LIBPCI_PROGRAMMERS=${yesNo (!stdenv.isDarwin)}"
+      "CONFIG_INTERNAL_X86=${yesNo (!(stdenv.isDarwin && stdenv.isx86_64))}"
+      "CONFIG_INTERNAL_DMI=${yesNo (!(stdenv.isDarwin && stdenv.isx86_64))}"
+      "CONFIG_RAYER_SPI=${yesNo (!(stdenv.isDarwin && stdenv.isx86_64))}"
```
This should be `!stdenv.isDarwin && stdenv.isx86_64`

funkeleinhorn replied:
Changed that!

felixsinger commented on pkgs/by-name/fl/flashprog/package.nix:
```diff
@@ -24,24 +26,36 @@ stdenv.mkDerivation rec {
+      "CONFIG_JLINK_SPI=${yesNo withJlink}"
+      "CONFIG_LINUX_GPIO_SPI=${yesNo withGpio}"
+      "CONFIG_ENABLE_LIBPCI_PROGRAMMERS=${yesNo (!stdenv.isDarwin)}"
+      "CONFIG_INTERNAL_X86=${yesNo (!(stdenv.isDarwin && stdenv.isx86_64))}"
```
No x86 internal programmer for x86? This should be `!stdenv.isDarwin && stdenv.isx86_64`

funkeleinhorn replied:
Changed that!

felixsinger commented on pkgs/by-name/fl/flashprog/package.nix:
```diff
@@ -24,24 +26,36 @@ stdenv.mkDerivation rec {
+      "CONFIG_LINUX_GPIO_SPI=${yesNo withGpio}"
+      "CONFIG_ENABLE_LIBPCI_PROGRAMMERS=${yesNo (!stdenv.isDarwin)}"
+      "CONFIG_INTERNAL_X86=${yesNo (stdenv.isLinux && stdenv.isx86_64)}"
+      "CONFIG_INTERNAL_DMI=${yesNo (!(stdenv.isDarwin && stdenv.isx86_64))}"
```
This should be `!stdenv.isDarwin && stdenv.isx86_64`

funkeleinhorn replied:
Fixed that!

felixsinger commented:
Fingers crossing that it passes the CI now! :crossed_fingers:
